infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/DiscoveryServiceEvent.java

```diff
@@ -5,20 +5,37 @@
 
 import org.corfudb.infrastructure.logreplication.proto.LogReplicationClusterInfo.TopologyConfigurationMsg;
 
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
 public class DiscoveryServiceEvent {
     DiscoveryServiceEventType type;
 
+    // Used by enforcing snapshot sync event
+    @Getter
+    private UUID eventUUID;
+
+    // Used by enforcing snapshot sync event
+    @Getter
+    CompletableFuture<UUID> cf;
```

Comment:
A specific name would be better, and looks like CompletableFuture<Void> is enough here.